tests/test_gmsd.py

```diff
@@ -102,12 +154,17 @@ def test_multi_scale_gmsd_fails_for_greyscale_tensors_chromatic_flag():
         loss(prediction, target)
 
 
-def test_multi_scale_gmsd_supports_custom_scale_weights(prediction: torch.Tensor, target: torch.Tensor):
-    try:
-        loss = MultiScaleGMSDLoss(scale_weights=[3., 4., 2., 1., 2.])
-        loss(prediction, target)
-    except Exception as e:
-        pytest.fail(f"Unexpected error occurred: {e}")
+def test_multi_scale_gmsd_supports_custom_weights(
+        prediction: torch.Tensor, target: torch.Tensor):
+    loss = MultiScaleGMSDLoss(scale_weights=[3., 4., 2., 1., 2.])
+    loss(prediction, target)
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason='No need to run test if there is no GPU.')
```

Comment:
I see this in many tests. I suggest using pytest fixtures:
1. Create conftest.py with
```
from typing import List, Any

import pytest
import torch

devices: List[str] = ["cpu"]
if torch.cuda.is_available():
    devices.append("cuda")


@pytest.fixture(params=devices)
def device(request: Any) -> Any:
    return request.param
```

2. Now, you can write tests with
```
def test_multi_scale_gmsd_supports_custom_weights_on_gpu(
        prediction: torch.Tensor, target: torch.Tensor, device: str):
    loss = MultiScaleGMSDLoss(scale_weights=[3., 4., 2., 1., 2.])
    loss(prediction.to(device), target.to(device))
```

Reply:
That's a good idea, but generally all testing pipeline in this lib should be changed, because now it's not unified from metric to metric. See #97 
So instead of patching it here I'll open new PR later and change everything at once.